test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -1630,9 +1630,6 @@ public void testBackupRestoreWorkflow() throws Exception {
         updateTopology(activeCorfuStore, DefaultClusterManager.OP_BACKUP);
         log.info("Change the topology!!!");
 
-        TimeUnit.SECONDS.sleep(shortInterval);
```

Comment:
yeah the code path is not interacted directly, and I was trying to achieve such workflow:
* Test method write to config table for OP_BACKUP
* Stream listener consumes update for OP_BACKUP, and invokes clusterManager's call back for applying new topology, and blocks until the new topology appears in DefaultClusterManager (DefaultClusterManager is registered in the stream listener)
* Test method write to config table for OP_ENFORCE_SNAPSHOT_FULL_SYNC (could happen before stream listener finishes consuming OP_BACKUP)
* Stream listener consumes update for OP_ENFORCE_SNAPSHOT_FULL_SYNC

Actually I think the second sleep (several lines below) could also be removed ... as we will block for the forced snapshot sync by checking table size

For the CountdownLatch method, if I understand correctly, it's also for blocking until topology change complete? but actually we didn't expose DefaultClusterManager to the test class, we only exposed some static fields like the OPs. So I think it will also work but could introduce more change...

Reply:
i was thinking of having a countdown latch or similar construct as well instead of the while loop.
I think conditionVariable is very apt here.

Everytime its OP_BACKUP, wait on the conditionVariable, and once the configId in the defaultClusterManager is updated, notify the conditionVariable.

https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Condition.html

Reply:
Changed to Condition, also for other topology config change cases. Thanks!